hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,373 @@
+        It reset any port to default port config as defined in
+        `default_port_config.j2`.
+        """
+        port_name = {'port_name': port}  # make sure the key here matches
+        # variable names used in the corresponding jinja templates.
+        jun = self._create_session()
+        err_list = []  # To avoid the issue of referencing before assigment.
+        # in case where only one of cfg.load fails.
+        rm_curr_port_config = """
+        delete interfaces {iface}
+        """.format(iface=port_name['port_name'])
+
+        base_config = self.dir_name\
+            + '/junos/jinja_templates/default_port_config.j2'
+
+        try:
+            with jun:
+                try:
+                    jun.cfg.load(rm_curr_port_config, format="set")
+                except ConfigLoadError as e1:
+                    err_list.append(e1)
+                try:
+                    jun.cfg.load(
+                            template_path=base_config,
+                            template_vars=port_name,
+                            format='text'
+                            )
+                except ConfigLoadError as e2:
+                    err_list.append(e2)
+                    return err_list
+
+                self._commit_config(jun)
+        except ConnectError as err:
+            print ("Cannot connect to device: {0}".format(err))
+            sys.exit(1)
+        except Exception as err:
+            print (err)
+
+    def modify_port(self, port, channel, network_id):
+        """ Changes vlan assignment to the port.
+        `node_connect_network` with 'vlan/native' flag:
+        enable port; set port to trunk mode; assign native_vlan;
+        remove default vlan
+        """
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if network_id is None:
+                self._remove_native_vlan(interface)
+            else:
+                self._set_native_vlan(interface, network_id)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if network_id is None:
+                self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert network_id == vlan_id
+                try:
+                    self._add_vlan_to_trunk(interface, vlan_id)
+                except VlanAddError as e:
+                    return e
+
+    def _set_native_vlan(self, port, network_id):
+        """Sets native vlan for a trunked port.
+        It enables the port, if it is the first vlan for the port.
+        """
+        port_info = self._interface_info(port)
+        self._set_commands(port, network_id)
+        set_native_vlan = self.set_native_vlan
+        if port_info[port]['trunk_mode']:
```

Comment:
`if not` and you can get rid of the else block

Reply:
done.